main.js

```diff
@@ -0,0 +1,286 @@
+      },
+      error: function (jqXHR, textStatus, errorThrown) {
+        alert(textStatus + ': City not found.');
+      }
+    })
+  }
+
+  var fetchByLocation = function (latitude, longitude) {
+    $.ajax({
+      method: 'GET',
+      url: 'https://api.openweathermap.org/data/2.5/weather?lat=' + latitude + '&lon=' + longitude + '&appid=' + API_key,
+      dataType: 'json',
+      success: function (data) {
+        addCurrentCityWeatherData(data);
+      },
+      error: function (jqXHR, textStatus, errorThrown) {
+        alert(textStatus + ': City not found.');
+      }
+    })
+  }
+
+  var fetchFiveDayForecast = function (currentCity) {
+    $.ajax({
+      method: 'GET',
+      url: 'https://api.openweathermap.org/data/2.5/forecast?q=' + currentCity + '&appid=' + API_key,
+      dataType: 'json',
+      success: function (data) {
+        addFiveDayCityWeatherData(data);
+      },
+      error: function (jqXHR, textStatus, errorThrown) {
+        console.log(errorThrown);
+      }
+    })
+  }
+
+  var fetchFiveDayForecastByLocation = function (latitude, longitude) {
+    $.ajax({
+      method: 'GET',
+      url: 'https://api.openweathermap.org/data/2.5/forecast?lat=' + latitude + '&lon=' + longitude + '&appid=' + API_key,
+      dataType: 'json',
+      success: function (data) {
+        addFiveDayCityWeatherData(data);
+      },
+      error: function (jqXHR, textStatus, errorThrown) {
+        console.log(errorThrown);
+      }
+    })
+  }
+
+  // Converts temperature to Fahrenheit
+  var tempToFahrenheit = function () {
+    var tempKelvin = currentCityWeatherData[0].temperature;
+    var tempFahrenheit = Math.floor(1.8 * (tempKelvin - 273) + 32);
+
+    currentCityWeatherData[0].temperature = tempFahrenheit.toString() + '°';
+  }
+
+  // converts date data to day of the week
+  var convertToDayOfWeek = function (forecastDataArray) {
+    var daysOfTheWeek = [ { 0: 'Sunday', 1: 'Monday', 2: 'Tuesday', 3: 'Wednesday', 4: 'Thursday', 5: 'Friday', 6: 'Saturday' } ];
+
+    forecastDataArray.forEach(e => {
+      var dayCode = e.date.getDay();
+      return e.date = daysOfTheWeek[0][dayCode];
+    })
+  }
+
+  var compressWeeklyWeatherData = function (forecastDataArray) {
+    // groups data by day of the week
+    var weeklyData = _.values(_.groupBy(forecastDataArray, 'date'));
+
+    fiveDayCityWeatherData = [];
+    
+    weeklyData.forEach(day => {
+      var weekday = day[0].date;
+      var highestTemp = 0;
+      var dailyConditions = [];
+      var dailyIcons = [];
+
+      // Populates the hottest temp of the day
+      for (let i = 0; i < day.length; i++) {
+        if (day[i].temp > highestTemp) {
+          highestTemp = day[i].temp;
+        }
+
+      dailyConditions.push(day[i].conditions);  
+      dailyIcons.push(day[i].weatherIcon);
+      }
+
+      // Converts temperature to Fahrenheit
+      var temp = Math.floor(1.8 * (highestTemp - 273) + 32);
+      temp = temp.toString() + '°';
```

Comment:
This is awesome, but I might separate out this "view" kind of stuff from this function.